File: dropwizard-core/src/test/java/com/yammer/dropwizard/jersey/params/tests/IntParamTest.java
package com.yammer.dropwizard.jersey.params.tests;

import com.yammer.dropwizard.jersey.params.IntParam;
import org.junit.Test;

import javax.ws.rs.WebApplicationException;
import javax.ws.rs.core.Response;

import static org.fest.assertions.api.Assertions.assertThat;
import static org.fest.assertions.api.Assertions.failBecauseExceptionWasNotThrown;

public class IntParamTest {
    @Test
    public void anIntegerReturnsAnInteger() throws Exception {
        final IntParam param = new IntParam("200");

        assertThat(param.get())
                .isEqualTo(200);
    }

    @Test
    @SuppressWarnings("ResultOfObjectAllocationIgnored")
    public void aNonIntegerThrowsAnException() throws Exception {
        try {
            new IntParam("foo");
            failBecauseExceptionWasNotThrown(WebApplicationException.class);
        } catch (WebApplicationException e) {
            final Response response = e.getResponse();

            assertThat(response.getStatus())
                    .isEqualTo(400);

            assertThat(response.getEntity())
                    .isEqualTo("\"foo\" is not a number.");
        }
    }
}

File: dropwizard-scala_2.9.1/src/test/scala/com/yammer/dropwizard/scala/params/tests/IntParamTest.scala
package com.yammer.dropwizard.scala.params.tests

import org.junit.Test
import javax.ws.rs.WebApplicationException
import com.simple.simplespec.Spec
import com.yammer.dropwizard.scala.params.IntParam

class IntParamTest extends Spec {

  class `A valid int parameter` {
    val param = IntParam("40")

    @Test def `has an int value` = {
      param.value.must(be(40))
    }
  }

  class `An invalid int parameter` {
    @Test def `throws a WebApplicationException with an error message` = {
      evaluating {
        IntParam("poop")
      }.must(throwAnExceptionLike {
        case e: WebApplicationException => {
          val response = e.getResponse
          response.getStatus.must(be(400))
          response.getEntity.must(be("Invalid parameter: poop (Must be an integer value.)"))
        }
      })
    }
  }

}
